src/FakeItEasy/Fake.of.T.cs

```diff
@@ -21,7 +21,7 @@ public class Fake<T> : IHideObjectMembers where T : class
         /// </summary>
         public Fake()
         {
-            this.FakedObject = (T)FakeAndDummyManager.CreateFake(typeof(T), null);
+            this.FakedObject = (T)FakeAndDummyManager.CreateFake(typeof(T), null, new LoopDetectingResolutionContext());

```

Comment:
Not related to your change, but is there a reason why we don't use the `CreateFake` method as in the other constructor?

Reply:
We could just pass an options builder that does nothing... Anyway, it's off topic for this PR